### ISSUE TITLE
Helm installation will not create default Jenkins instance if operator is in different namespace

### ISSUE THREAD

**Issue by r0bnet** [jenkinsci/kubernetes-operator#541]
**Describe the bug**
When following the documentation and installing the operator e.g. in the `jenkins` namespace then it will start to watch exactly this namespace:

```yaml
...
env:
- name: WATCH_NAMESPACE
  valueFrom:
    fieldRef:
      fieldPath: metadata.namespace
...
```

If you leave now a default Jenkins instance then it will be deployed in the `default` namespace which means that it will not be created since the operator watches `jenkins`.

```yaml
apiVersion: jenkins.io/v1alpha2
kind: Jenkins
metadata:
  name: jenkins
  namespace: default
...
```

All the information above is taken from the rendered chart itself (even when using `-n jenkins`).

Two approaches to fix it:
- Watch all namespaces (probably not)
- fix the namespace of the Jenkins instance in the chart to use the same one as the operator.

I guess the problem is that the template for the Jenkins instances sources the namespace from a specific value `jenkins.namespace` but not sure if that's the correct way.

```yaml
# namespace is the namespace where the resources will be deployed
# It's not recommended to use default namespace
# Create new namespace for jenkins (called e.g. jenkins)
namespace: default
```

Kubernetes version: 1.20.4
Jenkins Operator version: 
- Chart: 0.4.3
- Operator: 0.5.0

**Comment by prryb** [comment]
Thanks for reporting, I'll try to take a look into this.

**Comment by fatmcgav** [comment]
> Two approaches to fix it:
> 
> * Watch all namespaces (probably not)
> * fix the namespace of the Jenkins instance in the chart to use the same one as the operator.

Watching a list of namespaces could be beneficial, as it enables the splitting of Jenkins instances by team or project into difference namespaces for a few reasons, such as access control, resource management, accounting, etc...

**Comment by NesManrique** [comment]
I also hit this issue today. I see value on watching multiple namespaces as well. I actually need to deploy multiple jenkins for several purposes.

**Comment by prryb** [comment]
OK, here's what I've learned.

The `namespace` defined in `values.yaml` file and `-n` parameter to `helm` serve different purposes:
-  `namespace` in `values.yaml` is used to deploy Jenkins CR
- `helm -n {namespace}`  will be used as a namespace for the Operator itself.

So using namespace `jenkins` in `values.yaml` and supplying `-n operator` will result in an operator being created in `operator` namespace and a Jenkins instance in `jenkins` namespace. Then the operator should watch `jenkins` namespace.

Now, as @r0bnet noticed, there's a tiny bug in the helm chart, where the operator always will watch his own namespace. I'll be opening a PR to fix that.

As for watching multiple namespaces, currently, it is not supported. Right now it's best to have a 1:1 relationship between an instance of the operator and Jenkins. While it's possible to have one operator instance managing multiple Jenkins instances  (all in the same namespace), it may lead to some issues (e.g. only one instance would be backed up).
Adding support for that requires a major effort, but it is a very welcome addition. There's no ETA on that yet though. There are some architectural changes happening right now ([ROADMAP.md](https://github.com/jenkinsci/kubernetes-operator/blob/master/ROADMAP.md)) that need to be finished before that.

**Comment by KorusMateusz** [comment]
Fixed in Helm Chart 0.5.0
